test/framework/clusterctl/client.go

```diff
@@ -215,32 +221,72 @@ func ConfigCluster(ctx context.Context, input ConfigClusterInput) []byte {
 	return yaml
 }
 
-// ConfigClusterWithBinary uses clusterctl binary to run config cluster.
+// ConfigClusterWithBinary uses clusterctl binary to run config cluster or generate cluster.
+// NOTE: This func detects the clusterctl version and uses config cluster or generate cluster
+// accordingly. We can drop the detection when we don't have to support clusterctl v0.3.x anymore.
 func ConfigClusterWithBinary(_ context.Context, clusterctlBinaryPath string, input ConfigClusterInput) []byte {
-	log.Logf("clusterctl config cluster %s --infrastructure %s --kubernetes-version %s --control-plane-machine-count %d --worker-machine-count %d --flavor %s",
-		input.ClusterName,
-		valueOrDefault(input.InfrastructureProvider),
-		input.KubernetesVersion,
-		*input.ControlPlaneMachineCount,
-		*input.WorkerMachineCount,
-		valueOrDefault(input.Flavor),
-	)
+	log.Logf("Detect clusterctl version via: clusterctl version")
+
+	out, err := exec.Command(clusterctlBinaryPath, "version").Output()
+	Expect(err).ToNot(HaveOccurred(), "error running clusterctl version")
+	var clusterctlSupportsGenerateCluster bool
+	if strings.Contains(string(out), "Major:\"1\"") {
```

Comment:
Not really happy with this detection, but not sure what better alternatives would be

Maybe it's good enough given that we can drop it when we drop v0.3.x support.